daemon/cmd/daemon_main.go

```diff
@@ -1534,6 +1543,19 @@ func initEnv(cmd *cobra.Command) {
 			)
 		}
 	}
+
+	// Enabling NetfilterCompatibleMode increases BPF instruction count size, and may
+	// cause issues in kernels that have lower instruction complexity limit. Hence enabling,
+	// this feature only in kernels support higher instruction complexity limit (5.2 or later).
+	// NetfilterCompatibleMode feature can be enabled on all kernels, once we have support
+	// for adding new tail calls in host device[bpf_host].
+	// If kernel doesn't support higher instruction complexity limit, then disable NetfilterCompatibleMode.
+	if option.Config.NetfilterCompatibleMode {
+		if !probes.NewProbeManager().GetMisc().HaveLargeInsnLimit {
+			option.Config.NetfilterCompatibleMode = false
+			log.Warn("netfilter-compatible-mode is enabled only on 5.2 and higher versions of kernel.")
```

Comment:
It is not _always_ enabled on recent kernels.
```suggestion
			log.Warn("netfilter-compatible-mode requires kernel version 5.2 or higher.")
```